### PR TITLE
[shuffle] Add Message module example

### PR DESCRIPTION
**Pull request by tzakian**
This adds a `Message` module demo for Shuffle. This allows you to:
* Set a message on-chain; and
* If the set updates the message an event will be emitted saying what the message was changed from and what it was changed to.

You can compile this and generate transaction builders as follows (assuming you are in `shuffle/move/examples/Message`:
```
cargo run -p move-cli -- package --abi build
```

You can then build the transaction builders for this in typescript/javascript (again, assuming you are in `shuffle/move/examples/Message`):

```
cargo run -p transaction-builder-generator --  --language typescript \
    --module-name diemStdlib \
    --with-diem-types "../../../../testsuite/generate-format/tests/staged/diem.yaml" \
    --target-source-dir txn_builders .
```

This will place the generated typescript builders in `shuffle/move/examples/Message/txn_builders`

### CODE REVIEW

**Review by dimroc — APPROVED** [review]
LGTM but the doctests for `shuffle-custom-move-code` seem to be failing even though these changes are unrelated.

**Review by dimroc — APPROVED** [review]

File: shuffle/move/examples/Message/Move.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[package]
+name = "Message"
+version = "0.0.0"
+
+[addresses]
+MessageAddress = "0x2"
+
+[dependencies]
+MoveStdlib = { local = "../../../../language/move-stdlib", addr_subst = { "Std" = "0x1" }}
```
Comment (metajack) [inline shuffle/move/examples/Message/Move.toml:9]: Are we planning to have the stdlib inside the tool somehow? It seems odd we need to specify this.

File: shuffle/move/examples/Message/sources/set_message.move [hunk 2]
```diff
@@ -0,0 +1,7 @@
+script {
+    use MessageAddress::Message;
+
+    fun set_message(account: signer, message: vector<u8>) {
```
Comment (metajack) [inline shuffle/move/examples/Message/sources/set_message.move:4]: Can we not use script funs for this?

File: shuffle/move/examples/Message/Move.toml (hunk 1 above)
Comment (dimroc) [reply shuffle/move/examples/Message/Move.toml:9]: We are, but I believe due to some limitations with the Move package system, we're temporarily referencing it in the Move.toml. This unblocks us from forward progress, but ultimately I think this work item belongs to Move packaging.

File: shuffle/move/examples/Message/sources/set_message.move (hunk 2 above)
Comment (dimroc) [reply shuffle/move/examples/Message/sources/set_message.move:4]: We want to, but there are some limitations to Script Functions and the generated transaction builders. This was the stopgap.

**Pull request closed without merging**